clients/src/main/java/org/apache/kafka/clients/consumer/KafkaConsumer.java

```diff
@@ -795,6 +799,8 @@ private KafkaConsumer(ConsumerConfig config,
     public void subscribe(Collection<String> topics, ConsumerRebalanceListener listener) {
         acquire();
         try {
+            if (this.autoCommitEnabled)
```

Comment:
The call to `unsubscribe()` below already makes a call to `commitAsync`, so maybe it makes sense to move this to the `else` branch?